vignettes/ChromBand.Rmd

```diff
@@ -0,0 +1,379 @@
+
+```{r}
+hasSYM <- sapply(mget(featureNames(nsFiltered), symbolMap, ifnotfound=NA),
+                 function(x) length(x) > 0 && !is.na(x[1] ) )
+hasMAP <- sapply(mget(featureNames(nsFiltered), bandMap, ifnotfound=NA),
+                 function(x) length(x) > 0 && !is.na(x[1] ) )
+nsFiltered <- nsFiltered[hasSYM & hasMAP, ]
+```
+
+We define the *gene universe* to be the subset of genes that remain after this filtering.
+
+```{r defineGeneUniverse}
+affyUniverse <- featureNames(nsFiltered)
+entrezUniverse <- unlist(mget(affyUniverse, entrezMap))
+names(affyUniverse) <- entrezUniverse
+if (any(duplicated(entrezUniverse)))
+    stop("error in gene universe: can't have duplicate Entrez Gene Ids")
+```
+
+We assessed differential expression between the BCR/ABL and NEG groups using an empirical Bayes approach, as implemented in the software package limma [@limma], yielding an attenuated$t$-statistic for each gene.
+
+```{r parametric1}
+design <- model.matrix(~ 0 + nsFiltered$mol.biol)
+colnames(design) <- c("BCR/ABL", "NEG")
+contr <- c(1, -1)   ## NOTE: we thus have BCR/ABL w.r.t NEG
+fm1 <- lmFit(nsFiltered, design)
+fm2 <-contrasts.fit(fm1, contr)
+fm3 <- eBayes(fm2)
+ttestLimma <- topTable(fm3, number = nrow(fm3), adjust.method = "none")
+ttestLimma <- ttestLimma [featureNames(nsFiltered), ]
+tstats <- ttestLimma$t
+names(tstats) <- entrezUniverse [rownames(ttestLimma)]
+##
+```
+
+We used a $p$-value cutoff of $0.01$ to identify a list of potentially differentially expressed genes.
+
+```{r selectedSubset}
+ttestCutoff <- 0.01
+smPV <- ttestLimma$P.Value < ttestCutoff
+pvalFiltered <- nsFiltered [smPV, ]
+selectedEntrezIds <- unlist(mget(featureNames(pvalFiltered), entrezMap))
+##
+```
+
+# Methods
+
+There are two important features of gene sets based on chromosome bands: (1) the bands are nested hierarchically, and (2) they almost form a partition (for most species almost all genes appear in only one location in the genome). This naturally leads to two different dichotomies in approaches to testing: one is a top-down versus a bottom-up approach, and the other contrasts local tests with global tests. We use human chromosome 12 as an example to describe these approaches. Users will need to identify which of the approaches best align with their objectives and then make use of the software appropriately.
+
+   Conceptually one can start a sequential testing approach either at the most coarse level of organization (probably the level of the arm: p or q), or the most specific level (that of sub-sub-bands). In the top-down approach, one first tests the hypothesis of interest on a coarse level of organization, and if rejected, the next level of organization is considered. For example, we might first consider the band 12q21, and if that hypothesis is rejected, test each of the sub-bands 12q21.1, 12q21.2, and 12q21.3. If the hypothesis for 12q21.3 is rejected, then we may subsequently examine each of 12q21.31, 12q21.32, and 12q21.33.
```

Comment:
Remove the indents on paragraphs for this paragraph and others in the vignette.

Reply:
Please see the Pdf, that is how it is on the pdf. **You instructed us in the contribute article to keep it and make it look similar to the Pdf of the vignette.**   **then make up your mind if you want me to fix it or not.**
![Screen Shot 2023-04-01 at 6 25 05 AM](https://user-images.githubusercontent.com/57273086/229265499-e282a1f6-aa7f-41a1-baef-2c5e7acd87cb.png)